### PR TITLE
[IMPROVEMENT] Remove queued test of Blocked User

### PR DESCRIPTION
**Pull request by TheClashster**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [x] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [ ] I am an active contributor to the project.

---

### CODE REVIEW

File: mod_ci/controllers.py [hunk 1]
```diff
@@ -905,6 +905,24 @@ def blocked_users():
...
+            gh = GitHub(access_token=g.github['bot_token'])
+            repository = gh.repos(g.github['repository_owner'])(g.github['repository'])
+            if addUserForm.userID.data == payload['pull_request']['user']['id']:
+                tests = Test.query.filter(Test.pr_nr == payload['pull_request']['number']).all()
+                for test in tests:
+                    # Add canceled status only if the test hasn't started yet
+                    if len(test.progress) > 0:
+                        continue
+                    progress = TestProgress(test.id, TestStatus.canceled, "PR closed", datetime.datetime.now())
+                    g.db.add(progress)
```
Comment (satyammittal) [inline mod_ci/controllers.py]: you need to commit those db changes.

File: mod_ci/controllers.py [hunk 2]
```diff
@@ -905,6 +905,24 @@ def blocked_users():
             g.db.add(blocked_user)
             g.db.commit()
             flash('User blocked successfully.')
+            # Remove any queued pull request from blocked user
+            payload = request.get_json()
```
Comment (satyammittal) [inline mod_ci/controllers.py]: This request is not same as same Github response.

File: mod_ci/controllers.py [hunk 3]
```diff
@@ -905,6 +907,29 @@ def blocked_users():
...
             flash('User blocked successfully.')
+            # Remove any queued pull request from blocked user
+            gh = GitHub(access_token=g.github['bot_token'])
+            repository = gh.repos(g.github['repository_owner'])(g.github['repository'])
+            try:
+                # Getting all pull requests by blocked user on the repo
+                pulls = requests.get('https://api.github.com/repos/ccextractor/sample-platform/pulls/head={username}'
+                                     ).format(username=usernames[addUserForm.userID.data])
+                for pull in pulls:
+                    test = Test.query.filter(Test.pr_nr == pull['number']).all()
```
Comment (satyammittal) [inline mod_ci/controllers.py]: you are fetching "all" tests corresponding to a pr, You need to run a one more for loop for each test

File: mod_ci/controllers.py [hunk 4]
```diff
@@ -905,6 +907,29 @@ def blocked_users():
...
+                    g.db.add(progress)
+                    g.db.commit()
+                    repository.statuses(test.commit).post(
+                        state=Status.FAILURE,
+                        description="Tests canceled since user blacklisted",
+                        context="CI - {name}".format(name=test.platform.value),
+                        target_url=url_for('test.by_id', test_id=test.id, _external=True)
+                    )
+            except requests.exceptions.RequestException:
+                log.error('Pull Requests of Blocked User could not be fetched.')
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: You have the context available, so you can use g.log instead. That also makes you don't need to import it at the top.

File: mod_ci/controllers.py (hunk 3 above)
Comment (TheClashster) [reply mod_ci/controllers.py]: Ah, right.

Comment (TheClashster) [reply mod_ci/controllers.py]: Fixed 😃

File: mod_ci/controllers.py [hunk 5]
```diff
@@ -905,6 +907,29 @@ def blocked_users():
             g.db.add(blocked_user)
             g.db.commit()
             flash('User blocked successfully.')
+            # Remove any queued pull request from blocked user
+            gh = GitHub(access_token=g.github['bot_token'])
+            repository = gh.repos(g.github['repository_owner'])(g.github['repository'])
+            try:
+                # Getting all pull requests by blocked user on the repo
+                pulls = requests.get('https://api.github.com/repos/ccextractor/sample-platform/pulls/head={username}'
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: This url is not correct.

If possible, I'd prefer using the repository defined above. There's probably a possiblity to fetch it through there instead.

File: mod_ci/controllers.py [hunk 6]
```diff
@@ -905,6 +907,34 @@ def blocked_users():
             g.db.add(blocked_user)
             g.db.commit()
             flash('User blocked successfully.')
+            try:
+                # Remove any queued pull request from blocked user
+                gh = GitHub(access_token=g.github['bot_token'])
+                repository = gh.repos(g.github['repository_owner'])(g.github['repository'])
+                # Getting all pull requests by blocked user on the repo
+                pulls = gh.repos('ccextractor')('sample-platform').pulls.get()
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: Why do this if you have a `repository` object above that contains it all up to the `.pulls.get()` part?

File: mod_ci/controllers.py [hunk 7]
```diff
@@ -905,6 +907,34 @@ def blocked_users():
...
+                for pull in pulls:
+                    if pull['user']['id'] == addUserForm.userID.data:
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: If you invert this check and do a continue, you can unindent the rest of the code a bit more.

File: mod_ci/controllers.py [hunk 8]
```diff
@@ -905,6 +907,34 @@ def blocked_users():
...
+                                    state=Status.FAILURE,
+                                    description="Tests canceled since user blacklisted",
+                                    context="CI - {name}".format(name=test.platform.value),
+                                    target_url=url_for('test.by_id', test_id=test.id, _external=True)
+                                )
+                            except ApiError as a:
+                                log.error('Got an exception while posting to GitHub! Message: {message}'.format(
+                                    message=a.message))
+            except ApiError as a:
+                log.error('Pull Requests of Blocked User could not be fetched: {res}'.format(res=a.response))
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: Please make use of the `g.log` instead and remove the import at the top.

**Pull request merged**